tests/integration/targets/elb_application_lb_info/tasks/setup.yml

```diff
@@ -0,0 +1,84 @@
+- name: elb_application_lb_info setup
+  block:
+  - name: create VPC
+    ec2_vpc_net:
+      cidr_block: 10.228.228.0/22
+      name: '{{ resource_prefix }}_vpc'
+      state: present
+    register: vpc
+
+  - name: create internet gateway
+    ec2_vpc_igw:
+      vpc_id: '{{ vpc.vpc.id }}'
+      state: present
+      tags:
+        Name: '{{ resource_prefix }}'
+    register: igw
+
+  - name: create public subnet
+    ec2_vpc_subnet:
+      cidr: '{{ item.cidr }}'
+      az: '{{ aws_region}}{{ item.az }}'
+      vpc_id: '{{ vpc.vpc.id }}'
+      state: present
+      tags:
+        Public: '{{ item.public|string }}'
+        Name: '{{ item.public|ternary(''public'', ''private'') }}-{{ item.az }}'
+    with_items:
+    - cidr: 10.228.228.0/24
+      az: a
+      public: 'True'
+    - cidr: 10.228.229.0/24
+      az: b
+      public: 'True'
+    - cidr: 10.228.230.0/24
+      az: a
+      public: 'False'
+    - cidr: 10.228.231.0/24
+      az: b
+      public: 'False'
+    register: subnets
+
+  - ec2_vpc_subnet_info:
+      filters:
+        vpc-id: '{{ vpc.vpc.id }}'
+    register: vpc_subnets
+
+  - name: create list of subnet ids
+    set_fact:
+      alb_subnets: '{{ vpc_subnets|community.general.json_query(''subnets[?tags.Public == `True`].id'') }}'
```

Comment:
Not a blocker but nice to have, I'd like to eventually remove or reduce the CI dependency on community.general in the AWS collections so I've started switching to selectattr in tests that currently use json_query.
Example in tests/integration/targets/ec2_vpc_route_table/tasks/main.yml: https://github.com/ansible-collections/community.aws/pull/484/files